src/main/com/algos/src/datastructures/caches/LRUCache.java

```diff
@@ -32,27 +21,27 @@ public LRUCache(int cap) {
 
     private void setCapacity(int newCapacity) {
         checkCapacity(newCapacity);
-        for (int i = data.size(); i > newCapacity; i--) {
-            Entry<K, V> evicted = evict();
-            data.remove(evicted.getKey());
-        }
         this.cap = newCapacity;
+        
     }
 
     private Entry<K, V> evict() {
         if (head == null) {
-            throw new RuntimeException("cache cannot be empty!");
+            throw new RuntimeException("Cache cannot be empty!");
```

Comment:
**Security Issue**: Incorrect error handling could leak implementation details or system information, potentially aiding attackers in crafting further attacks. <br> **Fix**: Customize the error message to be less specific about the internal state or configuration of the cache system. <br> **Code Suggestion**: 
 ```
 -            throw new RuntimeException("Cache cannot be empty!");
 +            throw new RuntimeException("Cache operation failed due to empty state.");
 ```